internal/util/datastruc.go

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package util
+
+import (
+	"github.com/samber/lo"
+)
+
+// hashingListSubtract is a utility-function for subtracting a list from another list, using map's internal hashing function to do this more efficiently than lo.Difference or lo.Without (which only assume comparable, and thus run in quadratic time).
```

Comment:
This is a bit verbose for a utility function. Aim for a very concise single line summary if you can, followed by more detail if needed.

Something like...

```
// Subtract removes items in slice b from slice a, returning the result.
// Implemented using a map for greater efficiency on large slices than lo.Difference / lo.Without.
```